File: 조준장/level2/[3차] 파일명 정렬.py
def solution(files):
    answer = []
    file = []
    
    for i in range(len(files)):
        head = ""
        number = ""
        flag = False
        for j in files[i]:
            if j.isdigit():
                number += j
                flag = True
            else:
                if flag:
                    break
                if j.isalpha():
                    head += j.lower()
                    
                else:
                    head += j
                
        file.append([files[i], head, number, i])
    
    file.sort(key=lambda x:(x[1], int(x[2]), x[3]))

    for f in file:
        answer.append(f[0])

    return answer

File: 이훈일/Programmers/Level2/[3차] 파일명 정렬.java
import java.util.*;

class Solution {
    public String[] solution(String[] files) {
        Arrays.sort(files, new Comparator<String>(){
            @Override
            public int compare(String s1, String s2) {
                String[] s1Spl = strSplit(s1);
                String[] s2Spl = strSplit(s2);
                
                if(s1Spl[0].equals(s2Spl[0])) {
                    int num1 = Integer.parseInt(s1Spl[1]);
                    int num2 = Integer.parseInt(s2Spl[1]);
                    return num1 - num2;
                } else {
                    return s1Spl[0].compareTo(s2Spl[0]);
                }
            }
        });
            
        return files;   
    }
    
    public String[] strSplit(String str) {
        String lower = str.toLowerCase();
        String[] strArr = new String[3];
        strArr[0] = "";
        strArr[1] = "";
        strArr[2] = "";
        
        int idx = 0;
        for(int i = 0; i < str.length(); i++) {
            char word = lower.charAt(i);
            
            if(Character.isDigit(word)) {
                if(idx == 0) {
                    idx++;
                }
            } else {
                if(idx == 1) {
                    idx++;
                }
            }
            
            strArr[idx] += word;
        }
        
        return strArr;
    }
}
